pixelmatch.py

```diff
@@ -1,45 +1,105 @@
-DEFAULT_OPTIONS = {
-    "threshold": 0.1,  # matching threshold (0 to 1); smaller is more sensitive
-    "includeAA": False,  # whether to skip anti-aliasing detection
-    "alpha": 0.1,  # opacity of original image in diff ouput
-    "aa_color": [255, 255, 0],  # color of anti-aliased pixels in diff output
-    "diff_color": [255, 0, 0],  # color of different pixels in diff output
-    "diff_mask": False,  # draw the diff over a transparent background (a mask)
-}
+import functools
+from typing import Sequence, Union, List, Tuple
+
+
+ImageData = List[int]
+PILImageOrImageSequence = Union["Image", Sequence[float]]
+RGBTuple = Tuple[int, int, int]
+
+
+def pixelmatch(
+    img1: PILImageOrImageSequence,
+    img2: PILImageOrImageSequence,
```

Comment:
I think the support for PIL should inside a seperated function, or maybe a different import name(say `pixelmatch.contrib.PIL` .

Adding support for PIL directly inside the main `pixelmatch` function might cause some maintenance problem if we want to add support for other libraries ( say opencv or matplotlib ) as well.

Reply:
I think you're right on this one. Didn't even consider adding support for other images. The only benefit with this approach would be that you could mix ImageSequence and PILImage objects, together. However, if you are mixing like that I think it's fair to offload the responsibility of that to you.